packages/system/src/Stream/Stream/collectWhile.ts

```diff
@@ -0,0 +1,75 @@
+import type * as A from "../../Array"
+import * as T from "../../Effect"
+import { pipe } from "../../Function"
+import * as M from "../../Managed"
+import type { MutableArray } from "../../Mutable"
+import * as O from "../../Option"
+import * as Ref from "../../Ref"
+import * as Pull from "../Pull"
+import { Stream } from "./definitions"
+
+/**
+ * Maps an array until `none` is returned
+ */
+function collectWhileArray<X, Y>(
```

Comment:
Is there something in the lib to do that? The original collectWhile implementation uses a PartialFunction.

Reply:
I think this is actually missing but it is a good addition to the array module